lib/cadet_web/plug/check_admin.ex

```diff
@@ -0,0 +1,22 @@
+defmodule CadetWeb.Plug.CheckAdmin do
+  @moduledoc """
+  A plug that checks whether :current_user is an admin
+  """
+
+  import Plug.Conn
+
+  import Phoenix.Controller
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    if conn.assigns[:current_user] != nil && conn.assigns[:current_user].role == :admin do
```

Comment:
You can make an assumption that this Plug will be run after user is authenticated, otherwise it's programmer's error. `assert` is good for this.